cargo-shuttle/src/init.rs

```diff
@@ -0,0 +1,342 @@
+            );
+
+            if feature != Feature::Tower {
+                ShuttleInit::set_key_value_dependency_version(
+                    &self.framework.to_string(),
+                    &mut dependencies,
+                    &manifest_path,
+                    &url,
+                );
+            }
+
+            // TODO: We still need `sync_wrapper` for `axum`; this can be removed after service refactor #225
+            if feature == Feature::Axum {
+                ShuttleInit::set_key_value_dependency_version(
+                    "sync_wrapper",
+                    &mut dependencies,
+                    &manifest_path,
+                    &url,
+                );
+            }
+
+            if feature == Feature::Tower {
+                ShuttleInit::set_inline_table_dependency_version(
+                    "tower",
+                    &mut dependencies,
+                    &manifest_path,
+                    &url,
+                );
+
+                ShuttleInit::set_inline_table_dependency_features(
+                    "tower",
+                    &mut dependencies,
+                    vec!["full".to_string()],
+                );
+
+                ShuttleInit::set_inline_table_dependency_version(
+                    "hyper",
+                    &mut dependencies,
+                    &manifest_path,
+                    &url,
+                );
+
+                ShuttleInit::set_inline_table_dependency_features(
+                    "hyper",
+                    &mut dependencies,
+                    vec!["full".to_string()],
+                );
+            }
+        }
+
+        self.cargo_doc["dependencies"] = Item::Table(dependencies);
+
+        // Truncate Cargo.toml and write the updated `Document` to it
+        let mut cargo_toml = File::create(self.cargo_toml_path.clone())?;
+        cargo_toml.write_all(self.cargo_doc.to_string().as_bytes())?;
+
+        // Write boilerplate to `src/lib.rs` file
+        self.write_lib_file()?;
+
+        Ok(())
+    }
+
+    fn write_lib_file(&self) -> Result<()> {
+        let lib_code = self.get_boilerplate_code_for_framework();
+        if !lib_code.is_empty() {
+           let mut lib_file = File::create(self.lib_path.clone())?;
+            lib_file.write_all(lib_code.as_bytes())?;
+        }
+
+        Ok(())
+    }
+
+    fn get_boilerplate_code_for_framework(&self) -> &'static str {
+        match self.framework {
+            Framework::Axum => indoc! {r#"
+            use axum::{routing::get, Router};
+            use sync_wrapper::SyncWrapper;
+
+            async fn hello_world() -> &'static str {
+                "Hello, world!"
+            }
+
+            #[shuttle_service::main]
+            async fn axum() -> shuttle_service::ShuttleAxum {
+                let router = Router::new().route("/hello", get(hello_world));
+                let sync_wrapper = SyncWrapper::new(router);
+
+                Ok(sync_wrapper){}
+            }"#},
```

Comment:
Is it possible to put these in static/template files, rather than in the code?

Reply:
Oh good point! Yes that would be a great idea. I can just read the files and then write the content into `lib.rs`. And it makes testing easier. Thanks!

Reply:
Btw, should the `static/templates` files live inside `shuttle/cargo-shuttle/` directory?